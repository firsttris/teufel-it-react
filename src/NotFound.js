import React, { Component } from 'react';
import {
    Link
} from 'react-router-dom'
import Unicorn from './assets/images/NotFound/unicorn-small.png'

class NotFound extends Component {
    render() {
        return (
            <div>
                <img src={Unicorn} />
                <h2>Looks like something went wrong!</h2>
                <div>Page not found, try refreshing.</div>
                <div>If the problem persists feel free to contact us.</div>
                <div>
                    <div class="spacer">
                    </div>
                    <div><b>Contact:</b></div>
                    <div><a href="https://www.teufel-it.de">www.teufel-it.de</a></div>
                    <div><a href="mailto:info@teufel-it.de">info@teufel-it.de</a></div>
                    <br />
                    <Link to="/" class="btn btn-secondary">Start</Link>
                </div>
            </div>
        );
    }
}

export default NotFound;