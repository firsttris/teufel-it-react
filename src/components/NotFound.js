import React, { Component } from 'react';
import {
    Link
} from 'react-router-dom'
import Unicorn from './../assets/images/NotFound/unicorn-small.png'
import Lines from './../assets/images/backgrounds/lines.png'

class NotFound extends Component {
    render() {
        return (
            <div style={{ textAlign: 'center', fontFamily: 'monospace', backgroundImage: `url(${Lines})`, position: 'absolute', width: '100%', height: '100%', paddingTop: '100px'}}>
                <img src={Unicorn} alt="Not Found" style={{marginTop: '50px', marginBottom: '50px', height: '200px'}}/>
                <h2>Looks like something went wrong!</h2>
                <div>Page not found, try refreshing.</div>
                <div>If the problem persists feel free to contact us.</div>
                <div>
                    <div className="spacer" style={{ height: '80px' }}>
                    </div>
                    <div><b>Contact:</b></div>
                    <div><a href="https://www.teufel-it.de">www.teufel-it.de</a></div>
                    <div><a href="mailto:info@teufel-it.de">info@teufel-it.de</a></div>
                    <br />
                  <Link to="/" className="btn btn-secondary">Start</Link>
                </div>
            </div>
        );
    }
}

export default NotFound;