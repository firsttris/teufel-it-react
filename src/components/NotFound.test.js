import React from 'react';
import NotFound from './NotFound';
import renderer from 'react-test-renderer';
import {
  Router
} from 'react-router-dom'
import createHistory from 'history/createMemoryHistory'

test('Display Component', () => {
  const component = renderer.create(
    <Router history={createHistory()}>
      <NotFound></NotFound>
    </Router>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});