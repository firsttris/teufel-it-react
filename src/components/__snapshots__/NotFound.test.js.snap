// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Component 1`] = `
<div
  style={
    Object {
      "backgroundImage": "url(lines.png)",
      "fontFamily": "monospace",
      "height": "100%",
      "paddingTop": "100px",
      "position": "absolute",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <img
    alt="Not Found"
    src="unicorn-small.png"
    style={
      Object {
        "height": "200px",
        "marginBottom": "50px",
        "marginTop": "50px",
      }
    }
  />
  <h2>
    Looks like something went wrong!
  </h2>
  <div>
    Page not found, try refreshing.
  </div>
  <div>
    If the problem persists feel free to contact us.
  </div>
  <div>
    <div
      className="spacer"
      style={
        Object {
          "height": "80px",
        }
      }
    />
    <div>
      <b>
        Contact:
      </b>
    </div>
    <div>
      <a
        href="https://www.teufel-it.de"
      >
        www.teufel-it.de
      </a>
    </div>
    <div>
      <a
        href="mailto:info@teufel-it.de"
      >
        info@teufel-it.de
      </a>
    </div>
    <br />
    <a
      className="btn btn-secondary"
      href="/"
      onClick={[Function]}
    >
      Start
    </a>
  </div>
</div>
`;
